File: backend/app/schemas/agent.py
from datetime import datetime, timezone
from pydantic import BaseModel, ConfigDict


class AgentBase(BaseModel):
    name: str
    role: str
    status: str = "idle"
    current_task: str | None = None
    model: str
    last_heartbeat: str = ""
    success_rate: float = 0.0
    cost_today: float = 0.0


class AgentResponse(AgentBase):
    id: str

    model_config = ConfigDict(from_attributes=True)


class Meta(BaseModel):
    count: int
    timestamp: str = ""


class AgentListData(BaseModel):
    data: list[AgentResponse]
    meta: Meta

File: src/types/agent.ts
export interface Agent {
  id: string
  name: string
  role: 'coordinator' | 'researcher' | 'executor' | 'critic' | 'synthesizer'
  status: 'online' | 'idle' | 'working' | 'error'
  currentTask?: string
  model: string
  lastHeartbeat: string
  successRate: number
  costToday: number
}
